fix #835 :Link target of repo's GitHub Corner add new independent config `cornerExternalLinkTarget`.

Koooooo-7 opened the pull request:
fixes #843 

<!-- Please use English language -->
<!-- Please don't delete this template -->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

**Summary**
Link target of repo's GitHub Corner opens  repo in current window/tab.
Fix : Link target of repo's GitHub Corner according to externalLinkTarget config. #835
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.


**Other information:**

---

* [ ] DO NOT include files inside `lib` directory.

anikethsaha commented on src/core/render/index.js:
```diff
@@ -226,7 +226,7 @@ export function initRender(vm) {
   if (el) {
     if (config.repo) {
-      html += tpl.corner(config.repo)
+      html += tpl.corner(config.repo, config.externalLinkTarget)
```
as this is specific for GitHub corner, so `externalLinkTarget` name might not be appropriate. Better give it some specific name related to  github corner.

Koooooo-7 replied:
thx for ur suggestion, I see what ur mean.

In my opinion. I think it is just the settings/options  of  the `link target attribute` at the corner.
no matter what kind of link at the corner ( such as `github` , `gitlab`  maybe in the future with the corner icon customized...etc.)
it is not based on the link type, so I mean it is unnecessary to make it related to each link type.

BUT, if the corner is  just for `github` now and then, I agree to give a specific name likes `githubExternalLinkTarget ` to it.

AND, I will  update the doc soon after this.

anikethsaha replied:
`ExternalLinkTarget` might be confusing to others as it may tell that all the links present in the website will be in a new tab.

What do you think about `cornerExternalLinkTarge`

Koooooo-7 replied:
Agree ,  I will change it and update the doc soon.

Koooooo-7 replied:
Another question.
I remember that  `ExternalLinkTarget` is one of the [configuration](https://github.com/docsifyjs/docsify/blob/develop/docs/configuration.md#externallinktarget) in the docs now.  it works on those links which are in the page content.
```
externalLinkTarget
type: String
default: _blank
Target to open external links. Default '_blank' (new window/tab)

window.$docsify = {
  externalLinkTarget: '_self' // default: '_blank'
};
```
If we has signed another name of the corner, the corner would not be according to `externalLinkTarget` config. it would be a separate options.
Maybe we can think the `externalLinkTarget` is the **GLOBAL CONFIG** of both page content links and corner.

What's your opinion?

anikethsaha replied:
Current `externalLinkTarget` is for the links inside the markdown.
So I guess you can have the cornerExternalLinkTarget name or anything which is not conflicting with this one and makes some understanding just by reading the name 👍 

Also, do try using `externalLinkTarget` to `_self` and your config option to `_blank` and vice versa and checking if there is any conflict between these two.

these two should be independent. 
_Which I think its currently as of your PR but just check one more time for verification_

Koooooo-7 replied:
Okay, My currently PR is make `corner link` according on the `externalLinkTarget` config too, cuz I just add the `target` attribute in the corner's code.

I will try to make a new  `cornerExternalLinkTarget` as an  independent config for the corner and make sure it will not be conflicting  with the current `externalLinkTarget` config.

anikethsaha commented:
you need to update the docs as well with this .

anikethsaha commented:
Update the docs and squash the commits and then we are good to merge 👍

Koooooo-7 commented:
> Update the docs and squash the commits and then we are good to merge

@anikethsaha  I updated the docs and squashed the commits.Is there anything else I need to do?
